guides/docs/testing/testing_controllers.md

```diff
@@ -78,46 +134,79 @@ Create, show and update have more typical ways to fail because they need a way t
 Let's run the test:
 
 ```console
-$ mix test test/controllers/user_controller_test.exs
+$ mix test test/hello_web/controllers/user_controller_test.exs
 ```
 
 We get 8 failures that say "Not implemented" which is good. Our tests don't have blocks yet.
 
+### The first test
+
 Let's add our first test. We'll start with `index/2`.
 
 ```elixir
+# test/hello_web/controllers/user_controller_test.exs
+
 defmodule HelloWeb.UserControllerTest do
-  use HelloWeb.ConnCase, async: true
+  use HelloWeb.ConnCase
 
-  alias Hello.{Repo, Accounts.User}
+  alias Hello.Accounts
 
-  test "index/2 responds with all Users" do
-    users = [ User.changeset(%User{}, %{name: "John", email: "john@example.com"}),
-              User.changeset(%User{}, %{name: "Jane", email: "jane@example.com"}) ]
+  @user1_attrs %{email: "grobblefruit@example.org", name: "John", password: "surf and skate"}
```

Comment:
I was basing off what the automatic tests look like.  When you run phx.gen.json, the automatic tests have attributes as in my version.   This then allows testing by comparing with variables, rather than with hard-coded strings.  

Not my project, so I'll do what you recommend, but hard-coded strings are bad.